bitnami/jaeger/templates/collector/service.yml

```diff
@@ -0,0 +1,76 @@
+  {{- if and .Values.collector.service.clusterIP (eq .Values.collector.service.type "ClusterIP") }}
+  clusterIP: {{ .Values.collector.service.clusterIP }}
+  {{- end }}
+  {{- if .Values.collector.service.sessionAffinity }}
+  sessionAffinity: {{ .Values.collector.service.sessionAffinity }}
+  {{- end }}
+  {{- if .Values.collector.service.sessionAffinityConfig }}
+  sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" .Values.collector.service.sessionAffinityConfig "context" $) | nindent 4 }}
+  {{- end }}
+  {{- if or (eq .Values.collector.service.type "LoadBalancer") (eq .Values.collector.service.type "NodePort") }}
+  externalTrafficPolicy: {{ .Values.collector.service.externalTrafficPolicy | quote }}
+  {{- end }}
+  ports:
+    - name: grpc
+      port: {{ .Values.collector.service.ports.grpc }}
+      targetPort: {{ .Values.collector.containerPorts.grpc }}
+      {{- if and (or (eq .Values.collector.service.type "NodePort") (eq .Values.collector.service.type "LoadBalancer")) (not (empty .Values.collector.service.nodePorts.grpc)) }}
+      nodePort: {{ .Values.collector.service.nodePorts.grpc }}
+      {{- else if eq .Values.collector.service.type "ClusterIP" }}
+      nodePort: null
+      {{- end }}
+    - name: binary
+      port: {{ .Values.collector.service.ports.binary }}
+      targetPort: {{ .Values.collector.containerPorts.binary }}
+      {{- if and (or (eq .Values.collector.service.type "NodePort") (eq .Values.collector.service.type "LoadBalancer")) (not (empty .Values.collector.service.nodePorts.binary)) }}
+      nodePort: {{ .Values.collector.service.nodePorts.binary }}
+      {{- else if eq .Values.collector.service.type "ClusterIP" }}
+      nodePort: null
+      {{- end }}
+    - name: zipkin
+      port: {{ .Values.collector.service.ports.zipkin }}
+      targetPort: {{ .Values.collector.containerPorts.zipkin }}
+      {{- if and (or (eq .Values.collector.service.type "NodePort") (eq .Values.collector.service.type "LoadBalancer")) (not (empty .Values.collector.service.nodePorts.zipkin)) }}
+      nodePort: {{ .Values.collector.service.nodePorts.zipkin }}
+      {{- else if eq .Values.collector.service.type "ClusterIP" }}
+      nodePort: null
+      {{- end }}
+    - name: admin
+      port: {{ .Values.collector.service.ports.admin }}
+      targetPort: {{ .Values.collector.containerPorts.admin }}
+      {{- if and (or (eq .Values.collector.service.type "NodePort") (eq .Values.collector.service.type "LoadBalancer")) (not (empty .Values.collector.service.nodePorts.admin)) }}
+      nodePort: {{ .Values.collector.service.nodePorts.admin }}
+      {{- else if eq .Values.collector.service.type "ClusterIP" }}
+      nodePort: null
+      {{- end }}
+    {{- if .Values.collector.service.extraPorts }}
+    {{- include "common.tplvalues.render" (dict "value" .Values.collector.service.extraPorts "context" $) | nindent 4 }}
+    {{- end }}
+  selector: {{- include "common.labels.matchLabels" . | nindent 4 }}
```

Comment:
You need to set a selector that only selects the pods of the collector. Add below the component: collector label. Check other charts for examples (grafana-loki can help with that)